### PR TITLE
Server headers, FormData, some docs improvements and fix for bug in ChunkedResponse

### PR DESCRIPTION
**Pull request by michalpokusa**
PR does not contain any breaking changes, I suggest publishing it as 4.1.0.

⭐ Added:
- `FormData` class and `Request.form_data` for parsing POST body for `"application/x-www-form-urlencoded"`, `"multipart/form-data"`, `"text/plain"` content types.
- Example that shows new functionality.
- `Server.headers` for setting headers that are sent with every response, without need for explicitly adding them in constructor.
- EDIT1: `Server.poll()` return value that indicates the result of polling e.g. `NO_REQUEST` or `REQUEST_HANDLED_RESPONSE_SENT`.

🛠️ Updated:
- Multiple docstrings are more verbose.
- Fix: In `ChunkedResponse` sending empty chunk was breaking functionality, because it ended response prematurely.
- Updated docs.
- EDIT1: `Server.stopped` is set to `True` in constructor, it changes to `False` only after calling `.start()` `.server_forever()`.

🏗️ Refactor:
- Unified `query_params` and `form_data` to use `_IFieldStorage` for shared functionality.

### CODE REVIEW

**Comment by michalpokusa** [comment]
If possible, please test the impact of these changes to request processing speed. In theory there should not be much overhead, but in my testing I noticed a very big range (80ms-500ms). Although this might be caused by my WiFi and the connection itself, it we be great it anyone could eliminate the possibility that it is the code's fault.

Thanks in advance. 🙂

File: examples/httpserver_form_data.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+@server.route("/form", [GET, POST])
+def form(request: Request):
+    """
+    Serve a form with the given enctype, and display back the submitted value.
+    """
+    enctype = request.query_params.get("enctype", "text/plain")
+
+    if request.method == POST:
+        posted_value = request.data.get("something")
```
Comment (FoamyGuy) [inline examples/httpserver_form_data.py:50]: I got an exception here for request not having property `data` I think maybe this is supposed to be `form_data` instead?

Comment (michalpokusa) [reply examples/httpserver_form_data.py]: You are right, this should be `form_data`. Already fixed.

**Review by FoamyGuy — APPROVED** [review]
These changes look good to me. 

I've retested all modified examples on a Feather S3 TFT with the latest versions. 

Really great new functionality and improved docstrings. Thanks @michalpokusa!

**Pull request merged**